### PR TITLE
add hw-81-linked

### PR DESCRIPTION
**Pull request by TarasHryts**

### CODE REVIEW

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 1]
```diff
@@ -3,48 +3,190 @@
...
     @Override
     public void add(T value, int index) {
-
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node temp = getNodeByIndex(index);
+        if (index == 0 && size == 0) {
+            add(value);
+            size++;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @TarasHryts you can move this to the end of method

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 2]
```diff
@@ -3,48 +3,190 @@
...
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (T temp : list) {
+            add(temp);
+            size++;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @TarasHryts Do you need this? You have already increased the size in the method add :)

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 3]
```diff
@@ -3,48 +3,190 @@
...
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index > size || index < 0) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: what if index == 5 and size == 5?

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 2 above)
Comment (TarasHryts) [reply src/main/java/core/basesyntax/MyLinkedList.java]: my mistake

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 1 above)
Comment (TarasHryts) [reply src/main/java/core/basesyntax/MyLinkedList.java]: done

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 3 above)
Comment (TarasHryts) [reply src/main/java/core/basesyntax/MyLinkedList.java]: done

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 4]
```diff
@@ -3,48 +3,190 @@
...
+    public Node<T> last;
+
+    public MyLinkedList() {
+        this.first = null;
+        this.last = null;
+    }
+
+    private static class Node<T> {
+        private T element;
+        private Node previous;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]:
```suggestion
        private Node<T> previous;
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 5]
```diff
@@ -3,48 +3,190 @@
...
+        private Node next;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]:
```suggestion
        private Node<T> next;
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 6]
```diff
@@ -3,48 +3,190 @@
...
+            return last;
+        }
+        Node result = first;
+        for (int i = 1; i <= index; i++) {
+            result = result.next;
+        }
+        return result;
+    }
+
+    private Node getNodeByElement(T t) {
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @TarasHryts what about generics?)
```suggestion
    private Node<T> getNodeByElement(T t) {
```

File: src/main/java/core/basesyntax/MyLinkedList.java [hunk 7]
```diff
@@ -3,48 +3,190 @@
...
+        Node elementToSort = first;
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/MyLinkedList.java]: @TarasHryts Element to sort? Are sorting here?

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 1 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/MyLinkedList.java]: @TarasHryts have you moved it?

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 2 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/MyLinkedList.java]: @TarasHryts Have you removed it?

File: src/main/java/core/basesyntax/MyLinkedList.java (hunk 3 above)
Comment (boroda4436) [reply src/main/java/core/basesyntax/MyLinkedList.java]: @TarasHryts Please verify you have pushed new commit!!

**Pull request closed without merging**